gradient/commands/datasets.py

```diff
@@ -676,3 +685,94 @@ def update_status():
                         for pre_signed in pre_signeds:
                             update_status()
                             pool.put(self._delete, url=pre_signed.url)
+
+
+class ImportDatasetCommand(BaseDatasetsCommand):
+    def create_secret(self, key, value, expires_in=86400):
+        client = api_sdk.clients.SecretsClient(
+            api_key=self.api_key,
+            logger=self.logger,
+            ps_client_name=CLI_PS_CLIENT_NAME,
+        )
+
+        response = client.ephemeral(key, value, expires_in)
+        return response
+
+
+    def get_command_string(self, url):
+        return "%s %s /data/output" % (IMPORTER_COMMAND, url)
+
+
+    def get_command(self, url, http_auth):
+        
+        if url.scheme == 'git' or url.scheme == 's3':
+            return self.get_command_string(url.geturl())
+        
+        if url.scheme == 'https' or url.scheme == 'http':
+            if not http_auth:
+                return self.get_command_string(url.geturl())
+            http_auth_url = "https://${{HTTP_AUTH}}@%s" % url.path
+            return self.get_command_string(http_auth_url)
+
+        return ""
+    
+    def get_env_vars(self, url, secrets):
+        # http_auth, access_key, secret_key
+
+        if url.scheme == 'git':
+            return ""
+
+        if url.scheme == 's3' and secrets[S3_ACCESS_KEY] and secrets[S3_SECRET_KEY]:
+            access_key_secret = self.create_secret(S3_ACCESS_KEY, secrets[S3_ACCESS_KEY])
+            secret_key_secret = self.create_secret(S3_SECRET_KEY, secrets[S3_SECRET_KEY])
+
+            access_key_value = "secret:ephemeral:%s" % access_key_secret[S3_ACCESS_KEY]
+            secret_key_value = "secret:ephemeral:%s" % secret_key_secret[S3_SECRET_KEY]
+
+            return {
+                S3_ACCESS_KEY: access_key_value,
+                S3_SECRET_KEY: secret_key_value
+            }
+
+        if url.scheme == 'https' and secrets[HTTP_SECRET]:
+            http_auth_secret = self.create_secret(HTTP_SECRET, secrets[HTTP_SECRET])
+            return {
+                HTTP_SECRET: http_auth_secret
+            }
+        
+        return ""
+
+    def import_dataset(self, workflow):
+        client = api_sdk.clients.JobsClient(
+            api_key=self.api_key,
+            ps_client_name=CLI_PS_CLIENT_NAME,
+        )
+        client.create(**workflow)
+
+
+    def execute(self, cluster_id, machine_type, dataset_id, dataset_url, http_auth, access_key, secret_key):
+        workflow = {
+            "cluster_id": cluster_id,
+            "container": DATASET_IMPORTER_IMAGE,
+            "machine_type": machine_type,
+            "project": PROJECT_NAME,
+            "datasets": [{ "id": dataset_id, "name": "output", "output": True }],
+            "project_id": None
+        }
+
+        url = urlparse(dataset_url)
+
+        if url.scheme not in SUPPORTED_URL:
+            self.logger.log('Invalid URL format supported [ git | s3 | https | http ]: {}'.format(dataset_url))
```

Comment:
you can use the supported type in the string using join

Reply:
👍